README.md

```diff
@@ -46,6 +46,26 @@ watcher.on('delete', function (filepath, root) { console.log('file deleted', fil
 watcher.close();
 ```
 
+In addition to change events, the `watchman`-specific watcher may emit a
+`fresh_instance` event that signals the state got lost and we will not receive
+notifications for some of the changes. When this happens it is recommended to
+empty or reset any cache that tracks the state of the files. The event is
+immediately followed by change events for each file that does exist, that
+you can leverage to fill a cache again.
+
+For example, say we want to write an application that keeps track of file paths
+in a `Set`. Here is a possible way to handle it:
+
+```js
+function findFiles() { return new Set(glob('**/*.js')); }
+var files = findFiles();
```

Comment:
this should just be an empty `Set()` to start with (at least with the base watchman implementation), as watchman will tell you about all of these in the initial response

Reply:
The problem is that the documentation doesn't mention that, and I believe the other watchers don't do that (providing "add" events for all files on start), so it would be confusing for the example not to follow the usual pattern. As such I'd suggest keeping the example like so, what do you think? (I know this is also incorrect, because a file may have been concurrently removed between the call to `glob` and the call to `sane`… but the other watchers are also incorrect for the same reason.)